Assets/Shopify/UIToolkit/Components/VariantSelector.cs

```diff
@@ -0,0 +1,101 @@
+namespace Shopify.UIToolkit {
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Collections.Specialized;
+    using System.Linq;
+    using Shopify.Unity;
+
+    using UnityEngine;
+
+    /// <summary>
+    /// Converts the list of product variant and their respective selectable options that looks like:
+    /// [{
+    ///    "selectedOptions": [{
+    ///       "name": "optionName1",
+    ///       "value": "value1"
+    ///     }, {
+    ///       "name": "optionName2",
+    ///       "value": "value1"
+    ///     }]
+    /// },
+    /// {
+    ///    "selectedOptions": [{
+    ///       "name": "optionName1",
+    ///       "value": "value2"
+    ///     }, {
+    ///       "name": "optionName2",
+    ///       "value": "value2"
+    ///     }]
+    /// }]
+    /// into a map that looks like:
+    /// {
+    ///   "optionName1": ["value1", "value2", ...],
+    ///   "optionName2": ["value1", "value2", ...],
+    /// }
+    ///
+    /// </summary>
+    public class VariantSelector {
+        private List<ProductVariant> _variants;
+
+        public VariantSelector(List<ProductVariant> variants) {
+            _variants = variants;
+        }
+
+        public Dictionary<string, List<string>> AllOptions() {
+            return ValidOptionsForSelections(new Dictionary<string, string>());
+        }
+
+        public Dictionary<string, List<string>> ValidOptionsForSelections(Dictionary<string, string> selections) {
```

Comment:
The contents of this whole function are double indented.

Reply:
Thats.. weird. I'll fix!